From bb0d63b905485f5576e285e2de83add13b0cd9cf Mon Sep 17 00:00:00 2001
From: Shimin Zhou <shimingx.zhou@intel.com>
Date: Mon, 21 Sep 2015 15:05:02 +0800
Subject: [PATCH 419/429] usb: connect edison board automatically after turning
 C2 to C1

If turning switch from C2 to C1, the board can't be connected
automatically. That's a regression issue of EDISON-1921. Making
the event to send only once after rebooting can solve this issue.

Signed-off-by: Shimin Zhou <shimingx.zhou@intel.com>
Signed-off-by: Zhenming Zhao <zhenmingx.zhao@intel.com>
---
 drivers/usb/dwc3/dwc3-intel-mrfl.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/dwc3/dwc3-intel-mrfl.c b/drivers/usb/dwc3/dwc3-intel-mrfl.c
index ecd9b49a95c9..058647cfcf3a 100644
--- a/drivers/usb/dwc3/dwc3-intel-mrfl.c
+++ b/drivers/usb/dwc3/dwc3-intel-mrfl.c
@@ -21,6 +21,7 @@
 #define VERSION "2.10a"
 
 static int otg_id = -1;
+static int reboot_flag = 1;
 static int enable_usb_phy(struct dwc_otg2 *otg, bool on_off);
 static int dwc3_intel_notify_charger_type(struct dwc_otg2 *otg,
 		enum power_supply_charger_event event);
@@ -507,9 +508,11 @@ static int dwc3_intel_set_power(struct usb_phy *_otg,
 
 int dwc3_intel_enable_vbus(struct dwc_otg2 *otg, int enable)
 {
-	atomic_notifier_call_chain(&otg->usb2_phy.notifier,
-			USB_EVENT_VBUS, &enable);
-
+	if (reboot_flag) {
+		atomic_notifier_call_chain(&otg->usb2_phy.notifier,
+				USB_EVENT_VBUS, &enable);
+		reboot_flag = 0;
+	}
 	return 0;
 }
 
-- 
2.37.3

